File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int>ans;
        
        map<int , int>mp;
        for (int i = 0 ; i < nums.size(); i++)
        {
            int needed = target - nums[i];
            if (mp.find(needed) != mp.end())
            {
                ans = vector<int>({mp[needed] , i});
                break;
            }
            //cout << needed << endl;
            mp[nums[i]] = i;
        }
        return ans;
        
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer , Integer>mp = new HashMap<>();
        for (int i = 0 ; i < nums.length; i++)
        {
            
            if (mp.containsKey(nums[i]))
            {
                return new int[]{mp.get(nums[i]) , i};
            }
            int needed = target - nums[i];
            mp.put(needed , i);
        }
        return new int[]{};
    }
}